src/NServiceBus.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -20,9 +21,20 @@ public static void Fill(this IBasicProperties properties, OutgoingMessage messag
                 properties.CorrelationId = message.Headers[NServiceBus.Headers.CorrelationId];
             }
 
+            DelayDeliveryWith delayDeliveryWith;
+            DoNotDeliverBefore doNotDeliverBefore;
             DiscardIfNotReceivedBefore timeToBeReceived;
+            long delay = 0;
 
-            if (TryGet(deliveryConstraints, out timeToBeReceived) && timeToBeReceived.MaxTime < TimeSpan.MaxValue)
+            if (TryGet(deliveryConstraints, out delayDeliveryWith))
+            {
+                delay = Convert.ToInt64(Math.Round(delayDeliveryWith.Delay.TotalSeconds) * 1000);
+            }
+            else if (TryGet(deliveryConstraints, out doNotDeliverBefore))
+            {
+                delay = Convert.ToInt64(Math.Round((doNotDeliverBefore.At - DateTime.UtcNow).TotalSeconds) * 1000);
+            }
+            else if (TryGet(deliveryConstraints, out timeToBeReceived) && timeToBeReceived.MaxTime < TimeSpan.MaxValue)
```

Comment:
In ASB we look at `DeliveryConstaint` on operation is `DelayedDeliveryConstraint` and use `Time.UtcNow() + delay` for scheduled time. Otherwise, it's a `DoNotDeliverBefore` and we use the delay value as scheduled time. We also set `TimeToLive` on the message regardless of the delayed delivery.  ASB will expire the message and won't deliver it if TTBR has elapsed.